pkg/provider/azure_storageaccount.go

```diff
@@ -112,13 +112,19 @@ func (az *Cloud) getStorageAccounts(accountOptions *AccountOptions) ([]accountWi
 				}
 			}
 
-			if to.Bool(acct.IsHnsEnabled) != to.Bool(accountOptions.IsHnsEnabled) {
-				continue
+			if acct.AccountProperties == nil {
```

Comment:
`acct.AccountProperties` should not be nil, I think you need to fix the unit test in azure file driver

Reply:
I see ok.